MonoGame.Framework/MathHelper.cs

```diff
@@ -321,6 +322,46 @@ public static float WrapAngle(float angle)
 	    public static bool IsPowerOfTwo(int value)
 	    {
 	         return (value > 0) && ((value & (value - 1)) == 0);
-	    }
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Matrix"/>.
+        /// </summary>
+        public static Matrix ToMonoGame(this System.Numerics.Matrix4x4 matrix)
```

Comment:
Wondering if there needs to be copious doc in `Matrix`/`Vector*` structs etc to always mirror `Numerics` layout? These are not things that will ever change but mostly as a way to ensure across platforms/byte-order etc it should hold true. (Tests?)

Also, it's kind of implicit but `ref <struct>` doesn't cause an alloc (cool) but may need a doc comment explaining the gymnastics happening here to help the compiler use the `Numerics` without any allocs.. i.e. [this](https://learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/performance/#pass-and-return-by-reference)

Reply:
are you talking about docs to inform developers using the xml comments? or are you talking about the layout attribute to ensure the struct matches the numerics attribute explicitly? I may be missing something so please let me know but both are row major matrices so I believe the memory layout should match. Tests would be a great idea though, just to be sure.

Instead of `ToMonoGame(this System.Numerics.Matrix4x4 matrix)` it should be `ToMonoGame(this ref System.Numerics.Matrix4x4 matrix)` to ensure the clarity of whats expected right? I can add a better description to the xml doc as well if it feels needed.

Reply:
> are you talking about docs to inform developers using the xml comments? or are you talking about the layout attribute to ensure the struct matches the numerics attribute explicitly? I may be missing something so please let me know but both are row major matrices so I believe the memory layout should match. Tests would be a great idea though, just to be sure.

I meant both: xml comment and a layout attribute. especially people may not be aware that there is link between the two.   (note: am not a maintainer, so this is mostly a drive-by :) )


> Instead of `ToMonoGame(this System.Numerics.Matrix4x4 matrix)` it should be `ToMonoGame(this ref System.Numerics.Matrix4x4 matrix)` to ensure the clarity of whats expected right? I can add a better description to the xml doc as well if it feels needed.

Here, I meant that you are also avoiding extra allocs (because of `ref <struct>`) which is kinda nice - but to me it was not obvious at first glance and hence I had to double check dotnet docs :) simply adding that to the xml comment will be nice...